frontend/src/components/Login.tsx

```diff
@@ -0,0 +1,119 @@
+        axios.post(`http://localhost:8080/public/api/v1/auth/sign-in`, {
+            login: data.login,
+            password: data.password
+        }).then(response => {
+            localStorage.setItem("jwt", response.data.jwt);
+            navigate("/home");
+        }).catch(error => {
+            setData({
+                ...data,
+                showAlert: true,
+            });
+        });
+    };
+
+    const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setData({
+            ...data,
+            [event.target.name]: event.target.value
+        });
+    };
+
+    const handleClickShowPassword = () => {
+        setData({
+            ...data,
+            showPassword: !data.showPassword,
+        });
+    };
+
+    const handleMouseDownShowPassword = (event: React.MouseEvent) => {
+        event.preventDefault();
+    };
+
+    return (
+        <form onSubmit={handleSubmit} style={{width: '100%'}}>
+            <ThemeProvider theme={loginTheme}>
+                <Container>
+                    <TextField
+                        label="Логин"
+                        name="login"
+                        required
+                        InputLabelProps={{required: false}}
+                        fullWidth
+                        onChange={handleInput}
+                    />
+                    <FormControl fullWidth>
+                        <InputLabel>Пароль</InputLabel>
+                        <OutlinedInput
+                            label="Пароль"
+                            name="password"
+                            required
+                            type={data.showPassword ? "text" : "password"}
+                            onChange={handleInput}
+                            endAdornment={
+                                <InputAdornment position="end">
+                                    <IconButton
+                                        onClick={handleClickShowPassword}
+                                        onMouseDown={handleMouseDownShowPassword}
+                                        edge="end"
+                                    >
+                                        {data.showPassword ? <VisibilityOff /> : <Visibility />}
+                                    </IconButton>
+                                </InputAdornment>
+                            }
+                        />
+                    </FormControl>
+                    <Alert
+                        sx={{"width": "80%"}}
+                        severity="error"
+                        style={data.showAlert ? {"visibility": "visible"} : {"visibility": "hidden"}}
+                    >Неверный логин или пароль</Alert>
+                    <Button
+                        type="submit"
+                        fullWidth
+                    >
+                        Войти
+                    </Button>
+                </Container>
+            </ThemeProvider>
+        </form>
+    );
```

Comment:
It seems like you should split this large component in pieces. Google it, please

Reply:
This component contains many required fields. Splitting into files won't make it much smaller. It only complicates the logic